aixplain/modules/pipeline/asset.py

```diff
@@ -359,21 +319,31 @@ def run_async(
         payload = json.dumps(payload)
         call_url = f"{self.url}/{self.id}"
         logging.info(f"Start service for {name}  - {call_url} - {payload}")
-        r = _request_with_retry(
-            "post", call_url, headers=headers, data=payload
-        )
+        r = _request_with_retry("post", call_url, headers=headers, data=payload)
 
         resp = None
         try:
             resp = r.json()
-            logging.info(
-                f"Result of request for {name}  - {r.status_code} - {resp}"
-            )
+            logging.info(f"Result of request for {name}  - {r.status_code} - {resp}")
 
             poll_url = resp["url"]
             response = {"status": "IN_PROGRESS", "url": poll_url}
         except Exception:
             response = {"status": "FAILED"}
+            if r.status_code == 401:
```

Comment:
Such conditions may be added at the begging of the `try` (after line 325). In this case, the method may return the following `response`:

```python
return {"status": "FAILED", "error": error_msg}
```

This comment is valid for all the modified running methods.